awx/main/credential_plugins/azure_kv.py

```diff
@@ -3,6 +3,16 @@
 from django.utils.translation import ugettext_lazy as _
 from azure.keyvault import KeyVaultClient, KeyVaultAuthentication
 from azure.common.credentials import ServicePrincipalCredentials
+from msrestazure import azure_cloud
+
+
+# https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/azure_cloud.py
+clouds = [
+    vars(azure_cloud)[n]
+    for n in dir(azure_cloud)
+    if n.startswith("AZURE_") and n.endswith("_CLOUD")
+]
+default_cloud = vars(azure_cloud)["AZURE_PUBLIC_CLOUD"]
```

Comment:
how can you be sure that we will have `AZURE_PUBLIC_CLOUD` set in env?

Reply:
ah nm I'm reading https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/azure_cloud.py now and I get it

Reply:
:+1: yep - it doesn't come from the env, it's just the name of the constants they use.